### PR TITLE
Display save button on companies, add canCreate flag to companies/sites/site areas

### PR DESCRIPTION
**Pull request by Fonsu**
* Temporarily display 'save' button to fix bug
* Need to further discuss about the way to fetch/set `canCreate`

### CODE REVIEW

**Comment by jerome-benoit** [comment]
> * Temporarily display 'save' button to fix bug

I think by scratching your head a bit more, you can find a proper fix: 
`*ngIf="canUpdateCompany"` miss a condition to handle the update authorization, the proper fix is to find the right one.

**Comment by LucasBrazi06** [comment]
@Fonsu As discussed via email, pass the available canCreate, canUpdate that you have in the list to the pop-up and use them.

**Comment by LucasBrazi06** [comment]
@Fonsu Any news on this one?

**Comment by Fonsu** [comment]
> @Fonsu Any news on this one?

I'm done with companies, can you take a look? At the moment I'm also replicating these changes to Sites and Site Areas, need to check if this approach will cover all other use cases. 

`createWithAuthorizations` is a temporary method to avoid changing 10+ create actions with potential regressions but my idea would be to remove the original `create` method eventually, once all use cases are covered and double checked.

**Comment by LucasBrazi06** [comment]
> > @Fonsu Any news on this one?
> 
> I'm done with companies, can you take a look? At the moment I'm also replicating these changes to Sites and Site Areas, need to check if this approach will cover all other use cases.
> 
> `createWithAuthorizations` is a temporary method to avoid changing 10+ create actions with potential regressions but my idea would be to remove the original `create` method eventually, once all use cases are covered and double checked.

If you want me to take a look, move your issue to To be reviewed ;-)

File: src/app/pages/organization/companies/company/company.dialog.component.ts [hunk 1]
```diff
@@ -5,16 +5,21 @@ import { Utils } from '../../../../utils/Utils';
...
   @ViewChild('appRef') public appRef!: CompanyComponent;
   public companyID!: string;
+  public canCreateCompany: boolean;
 
   public constructor(
     public dialogRef: MatDialogRef<CompanyDialogComponent>,
-    @Inject(MAT_DIALOG_DATA) data: string) {
-    this.companyID = data;
+    @Inject(MAT_DIALOG_DATA) public data: boolean | string, ) {
+    if(typeof data === 'boolean'){
```
Comment (LucasBrazi06) [inline src/app/pages/organization/companies/company/company.dialog.component.ts:18]: You should pass both canCreate and canUpdate in the data.
Make data a Json document which contains both flags + the companyID.

File: src/app/pages/organization/companies/company/company.component.ts [hunk 2]
```diff
@@ -24,6 +24,7 @@ import { Utils } from '../../../../utils/Utils';
 })
 export class CompanyComponent implements OnInit {
   @Input() public currentCompanyID!: string;
+  @Input() public canCreateCompany: boolean;
```
Comment (LucasBrazi06) [inline src/app/pages/organization/companies/company/company.component.ts:27]: Pass also canUpdateCompany.
No need to get it from the company in the HTTP request now.

File: src/app/pages/organization/companies/list/companies-list-table-data-source.ts [hunk 3]
```diff
@@ -201,7 +203,10 @@ export class CompaniesListTableDataSource extends TableDataSource<Company> {
       // Add
       case CompanyButtonAction.CREATE_COMPANY:
         if (actionDef.action) {
-          (actionDef as TableCreateCompanyActionDef).action(CompanyDialogComponent, this.dialog, this.refreshData.bind(this));
+          const authorizations: AuthorizationActions = {
+            canCreate: this.canCreateCompany
```
Comment (LucasBrazi06) [inline src/app/pages/organization/companies/list/companies-list-table-data-source.ts:207]: Add also canUpdate.

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]
Good approach for canCreate.
I would use also the same approach with canUpdate,
Check my comments.

**Comment by LucasBrazi06** [comment]
@Fonsu I finally merged it to have a better readability on the next changes.
I tested it and I didn't find any showstopper.
Other devs will also test their code with these changes

**Pull request merged**